server.go

```diff
@@ -1811,77 +1656,11 @@ out:
 			continue
 		}
 
-		// Only try connect to more peers if we actually need more.
-		if !state.NeedMoreOutbound() || len(cfg.ConnectPeers) > 0 ||
-			atomic.LoadInt32(&s.shutdown) != 0 {
-			state.forPendingPeers(func(sp *serverPeer) {
-				srvrLog.Tracef("Shutdown peer %s", sp)
-				sp.Disconnect()
-			})
-			continue
-		}
-		tries := 0
-		for state.NeedMoreOutbound() &&
-			state.NeedMoreTries() &&
-			atomic.LoadInt32(&s.shutdown) == 0 {
-			addr := s.addrManager.GetAddress("any")
-			if addr == nil {
-				break
-			}
-			key := addrmgr.GroupKey(addr.NetAddress())
-			// Address will not be invalid, local or unroutable
-			// because addrmanager rejects those on addition.
-			// Just check that we don't already have an address
-			// in the same group so that we are not connecting
-			// to the same network segment at the expense of
-			// others.
-			if state.outboundGroups[key] != 0 {
-				break
-			}
-
-			// Check that we don't have a pending connection to this addr.
-			addrStr := addrmgr.NetAddressKey(addr.NetAddress())
-			if _, ok := state.pendingPeers[addrStr]; ok {
-				continue
-			}
-
-			tries++
-			// After 100 bad tries exit the loop and we'll try again
-			// later.
-			if tries > 100 {
-				break
-			}
-
-			// XXX if we have limited that address skip
-
-			// only allow recent nodes (10mins) after we failed 30
-			// times
-			if tries < 30 && time.Now().Sub(addr.LastAttempt()) < 10*time.Minute {
-				continue
-			}
-
-			// allow nondefault ports after 50 failed tries.
-			if fmt.Sprintf("%d", addr.NetAddress().Port) !=
-				activeNetParams.DefaultPort && tries < 50 {
-				continue
-			}
-
-			tries = 0
-			sp := s.newOutboundPeer(addrStr, false)
-			if sp != nil {
-				go s.peerConnHandler(sp)
-				state.pendingPeers[sp.Addr()] = sp
-			}
-		}
-
-		// We need more peers, wake up in ten seconds and try again.
-		if state.NeedMoreOutbound() {
-			time.AfterFunc(10*time.Second, func() {
-				s.wakeup <- struct{}{}
-			})
-		}
+		// TODO: Connect to more peers - this is now managed by connmgr.
```

Comment:
The checks have been restored but I'm not sure if keeping the old `tries` loop is a good way to handle this. For reference:

https://github.com/btcsuite/btcd/pull/653/commits/54cad6a05c16e047235be05b895fa16791094a21#diff-34c6b408d72845d076d47126c29948d1L2327